luxonis_ml/data/datasets/annotation.py

```diff
@@ -231,25 +254,76 @@ def get_value(self) -> Dict[str, Any]:
             "counts": rle["counts"].decode("utf-8"),
         }
 
-    def to_numpy(self, _: Dict[str, int], width: int, height: int) -> np.ndarray:
+    def to_numpy(
+        self, _: Dict[str, int], width: int, height: int
+    ) -> npt.NDArray[np.bool_]:
         assert isinstance(self.counts, bytes)
-        return mask_util.decode({"counts": self.counts, "size": [height, width]})
+        return mask_util.decode(
+            {"counts": self.counts, "size": [height, width]}
+        ).astype(np.bool_)
+
+
+class MaskSegmentationAnnotation(SegmentationAnnotation):
+    """Pixel-wise binary segmentation mask.
+
+    @type mask: npt.NDArray[np.bool_]
+    @ivar mask: The segmentation mask as a numpy array. The mask must be 2D and must be
+        castable to a boolean array.
+    """
+
+    type_: Literal["mask"] = Field("mask", alias="type")
+    mask: npt.NDArray[np.bool_]
+
+    model_config = ConfigDict(extra="forbid", arbitrary_types_allowed=True)
+
+    @model_validator(mode="before")
+    @classmethod
+    def _convert_rle(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        if "mask" in values:
+            return values
+
+        if "width" not in values or "height" not in values or "counts" not in values:
```

Comment:
Why do we allow/check for "rle" type of annotations in the "mask" annotation type?

Reply:
That's for internal conversion back from RLE. Basically, the flow is `mask (user) -> RLE (internal) -> parquet (internal) -> RLE (internal) -> mask (user)`